### PR TITLE
updated warnings for a) ImageMagick and b) Conda

### PR DESCRIPTION
**Pull request by huckl3b3rry87**

### CODE REVIEW

**Comment by mkborregaard** [comment]
@huckl3b3rry87 and @oschulz , what is the relationship between this and https://github.com/JuliaPlots/Plots.jl/pull/782?

**Comment by oschulz** [comment]
#782 keeps the QT warning on Linux in, but only if Conda is actually managing packages (because otherwise, `Conda._installed_packages_dict()` will initialize Conda and install lot's of base packages, even though the user is using an external conda, e.g. via Anaconda). I don't know about the warning itself, or whether it's still relevant.

#785 takes the QT check/warning out completely - if it's no longer necessary, then that's preferable to #782, of course.

**Comment by huckl3b3rry87** [comment]
@oschulz thank you for answering @mkborregaard 's question.

I was the one that actually initially created this warning in https://github.com/JuliaPlots/Plots.jl/pull/651, but it is no longer an issue for me. Everything seems to be working fin now with the latest version of qt on Linux.

**Comment by oschulz** [comment]
Thanks, @huckl3b3rry87.

@mkborregaard, I closed #782 and recommend to merge this PR soon, instead.

File: src/animation.jl [hunk 1]
```diff
@@ -80,7 +80,9 @@ function buildanimation(animdir::AbstractString, fn::AbstractString;
 
     catch err
         warn("""Tried to create gif using convert (ImageMagick), but got error: $err
-            ImageMagick can be installed by executing `Pkg.add("ImageMagick")`
+            ImageMagick needs to be installed in julia and on your machine:
+              1) in julia>> `Pkg.add("ImageMagick")`
+              2) on your machine (in terminal run): sudo apt install imagemagick
```
Comment (musm) [inline src/animation.jl]: not true on windows

Comment (huckl3b3rry87) [inline src/animation.jl]: there does not need to be a separate install for imagemagick on windows? i.e. you just need to run add the package in julia?

Comment (musm) [inline src/animation.jl]: line 2) does not apply is what I meant

Comment (timholy) [inline src/animation.jl]: Theoretically it shouldn't be needed on any platform. Did you experience something different, @huckl3b3rry87?

**Comment by huckl3b3rry87** [comment]
Tim, that is what I was thinking, but yes imagemagick did not work until I
installed it on Ubuntu.

On Apr 29, 2017 5:31 AM, "Tim Holy" <notifications@github.com> wrote:

*@timholy* commented on this pull request.
------------------------------

In src/animation.jl
<https://github.com/JuliaPlots/Plots.jl/pull/785#discussion_r114049864>:

> @@ -80,7 +80,9 @@ function buildanimation(animdir::AbstractString, fn::AbstractString;

     catch err
         warn("""Tried to create gif using convert (ImageMagick), but
got error: $err
-            ImageMagick can be installed by executing `Pkg.add("ImageMagick")`
+            ImageMagick needs to be installed in julia and on your machine:
+              1) in julia>> `Pkg.add("ImageMagick")`
+              2) on your machine (in terminal run): sudo apt install
imagemagick

Theoretically it shouldn't be needed on any platform. Did you experience
something different, @huckl3b3rry87 <https://github.com/huckl3b3rry87>?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/JuliaPlots/Plots.jl/pull/785#discussion_r114049864>,
or mute
the thread

**Comment by timholy** [comment]
If you can recreate what happened when you tried to install, that would be great. [These lines](https://github.com/JuliaIO/ImageMagick.jl/blob/5d5b56d6bc38eca209cbd5a3593050e998e10876/deps/build.jl#L27-L34) theoretically should automatically install it, but perhaps the package name is not wide enough for different Ubuntu versions? What version are you running?

**Comment by oschulz** [comment]
@huckl3b3rry87: Could you maybe split this PR?

* One PR to remove the obsolete Conda/QT warning and merge it quickly (it causes an undesired and highly annoying Conda initialization for Anaconda users like me).

* One PR regarding the ImageMagick warning, as it seems to require a bit more analysis.

**Comment by mkborregaard** [comment]
I don't think we need to give installation instructions for ImageMagick - if you can just replace that line with "make sure the ImageMagick package is installed" or something I'll merge this.

**Comment by huckl3b3rry87** [comment]
@timholy I recreated my issue:
```
febbo@febbo-HP-ZBook-17-G2:~/.julia/v0.5$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.1 (2017-03-05 13:25 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using Plots

julia> p = plot([sin,cos],zeros(0),leg=false)

julia> anim = Animation()
Plots.Animation("/tmp/tmpEzLhES",String[])

julia> for x = linspace(0,10π,100)
           push!(p,x,Float64[sin(x),cos(x)])
           frame(anim)
       end

julia> gif(anim,"test",fps=10)
WARNING: Tried to create gif using convert (ImageMagick), but got error: could not spawn `convert -delay 10 -loop 0 /tmp/tmpEzLhES/*.png -alpha off /home/febbo/.julia/v0.5/test`: no such file or directory (ENOENT)
ImageMagick can be installed by executing `Pkg.add("ImageMagick")`
Will try ffmpeg, but it's lower quality...)
ERROR: failed process: Process(`ffmpeg -v 0 -framerate 10 -loop 0 -i /tmp/tmpEzLhES/%06d.png -y /home/febbo/.julia/v0.5/test`, ProcessExited(1)) [1]
 in pipeline_error(::Base.Process) at ./process.jl:616
 in run(::Cmd) at ./process.jl:592
 in #buildanimation#348(::Int64, ::Int64, ::Function, ::String, ::String) at /home/febbo/.julia/v0.5/Plots/src/animation.jl:87
 in (::Plots.#kw##buildanimation)(::Array{Any,1}, ::Plots.#buildanimation, ::String, ::String) at ./<missing>:0
 in #gif#345(::Array{Any,1}, ::Function, ::Plots.Animation, ::String) at /home/febbo/.julia/v0.5/Plots/src/animation.jl:58
 in (::Plots.#kw##gif)(::Array{Any,1}, ::Plots.#gif, ::Plots.Animation, ::String) at ./<missing>:0

julia> Pkg.status("ImageMagick")
 - ImageMagick                   0.2.4

julia> 
febbo@febbo-HP-ZBook-17-G2:~/.julia/v0.5$ sudo apt install imagemagick
[sudo] password for febbo: 
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  imagemagick-6.q16 libnetpbm10 netpbm
Suggested packages:
  imagemagick-doc autotrace enscript gimp gnuplot grads graphviz hp2xx html2ps
  libwmf-bin mplayer povray radiance texlive-base-bin transfig ufraw-batch
The following NEW packages will be installed:
  imagemagick imagemagick-6.q16 libnetpbm10 netpbm
0 upgraded, 4 newly installed, 0 to remove and 71 not upgraded.
Need to get 1,487 kB of archives.
After this operation, 5,262 kB of additional disk space will be used.
Do you want to continue? [Y/n] Y
Get:1 http://us.archive.ubuntu.com/ubuntu xenial-updates/main amd64 imagemagick-6.q16 amd64 8:6.8.9.9-7ubuntu5.6 [387 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu xenial-updates/main amd64 imagemagick amd64 8:6.8.9.9-7ubuntu5.6 [44.8 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu xenial/main amd64 libnetpbm10 amd64 2:10.0-15.3 [58.1 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu xenial/main amd64 netpbm amd64 2:10.0-15.3 [997 kB]
Fetched 1,487 kB in 0s (3,243 kB/s)
Selecting previously unselected package imagemagick-6.q16.
(Reading database ... 222062 files and directories currently installed.)
Preparing to unpack .../imagemagick-6.q16_8%3a6.8.9.9-7ubuntu5.6_amd64.deb ...
Unpacking imagemagick-6.q16 (8:6.8.9.9-7ubuntu5.6) ...
Selecting previously unselected package imagemagick.
Preparing to unpack .../imagemagick_8%3a6.8.9.9-7ubuntu5.6_amd64.deb ...
Unpacking imagemagick (8:6.8.9.9-7ubuntu5.6) ...
Selecting previously unselected package libnetpbm10.
Preparing to unpack .../libnetpbm10_2%3a10.0-15.3_amd64.deb ...
Unpacking libnetpbm10 (2:10.0-15.3) ...
Selecting previously unselected package netpbm.
Preparing to unpack .../netpbm_2%3a10.0-15.3_amd64.deb ...
Unpacking netpbm (2:10.0-15.3) ...
Processing triggers for bamfdaemon (0.5.3~bzr0+16.04.20160824-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for gnome-menus (3.13.3-6ubuntu3.1) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu5.1) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.15-0ubuntu1) ...
Processing triggers for man-db (2.7.5-1) ...
Processing triggers for libc-bin (2.23-0ubuntu7) ...
/sbin/ldconfig.real: "$HOME/knitro-10.1.2-z-Linux-64/lib:$LD_LIBRARY_PATH" is not a known library type
Setting up imagemagick-6.q16 (8:6.8.9.9-7ubuntu5.6) ...
Setting up imagemagick (8:6.8.9.9-7ubuntu5.6) ...
update-alternatives: using /usr/bin/compare-im6 to provide /usr/bin/compare (compare) in auto mode
update-alternatives: using /usr/bin/animate-im6 to provide /usr/bin/animate (animate) in auto mode
update-alternatives: using /usr/bin/convert-im6 to provide /usr/bin/convert (convert) in auto mode
update-alternatives: using /usr/bin/composite-im6 to provide /usr/bin/composite (composite) in auto mode
update-alternatives: using /usr/bin/conjure-im6 to provide /usr/bin/conjure (conjure) in auto mode
update-alternatives: using /usr/bin/import-im6 to provide /usr/bin/import (import) in auto mode
update-alternatives: using /usr/bin/identify-im6 to provide /usr/bin/identify (identify) in auto mode
update-alternatives: using /usr/bin/stream-im6 to provide /usr/bin/stream (stream) in auto mode
update-alternatives: using /usr/bin/display-im6 to provide /usr/bin/display (display) in auto mode
update-alternatives: using /usr/bin/montage-im6 to provide /usr/bin/montage (montage) in auto mode
update-alternatives: using /usr/bin/mogrify-im6 to provide /usr/bin/mogrify (mogrify) in auto mode
Setting up libnetpbm10 (2:10.0-15.3) ...
Setting up netpbm (2:10.0-15.3) ...
Processing triggers for libc-bin (2.23-0ubuntu7) ...
/sbin/ldconfig.real: "$HOME/knitro-10.1.2-z-Linux-64/lib:$LD_LIBRARY_PATH" is not a known library type
febbo@febbo-HP-ZBook-17-G2:~/.julia/v0.5$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.1 (2017-03-05 13:25 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> using Plots

julia> p = plot([sin,cos],zeros(0),leg=false)

julia> anim = Animation()
Plots.Animation("/tmp/tmp0Uy3F1",String[])

julia> for x = linspace(0,10π,100)
           push!(p,x,Float64[sin(x),cos(x)])
           frame(anim)
       end

julia> gif(anim,"test",fps=10)
INFO: Saved animation to /home/febbo/.julia/v0.5/test
Plots.AnimatedGif("/home/febbo/.julia/v0.5/test")

julia> 
```

**Comment by huckl3b3rry87** [comment]
@mkborregaard if the message just says "make sure the ImageMagick package is installed," then people may just make sure it is installed in julia (like I did and be confused for weeks until they put imagemagick on their machine).

@timholy I am running Ubuntu 16.04 ( and have recreated this on two machines).

@oschulz the branches for this pull request are on another machine. Also, I have never split a pull request. Would that just involve creating two new pull requests?

**Comment by timholy** [comment]
Now I understand better, thanks. The issue is that ImageMagick.jl installs [just libmagickwand](https://github.com/JuliaIO/ImageMagick.jl/blob/master/deps/build.jl) but something here is trying to use the command-line interface. Just use an `fps` option to `save` directly: https://github.com/JuliaIO/ImageMagick.jl/blob/5d5b56d6bc38eca209cbd5a3593050e998e10876/test/constructed_images.jl#L254-L256

**Comment by oschulz** [comment]
> Would that just involve creating two new pull requests?

No, you could use this one for the one issue, and create a new one for the other. Obviously, you'll have to split this branch on your machine split the other change out into another branch.

**Comment by oschulz** [comment]
> Would that just involve creating two new pull requests?

@huckl3b3rry87: If you like, I can create a new PR to remove the obsolete Conda/QT warning. Then you can rebase this one on top of it.

**Comment by mkborregaard** [comment]
@timholy ah, so that indicates this is indeed a Plots issue. The bit of code https://github.com/JuliaPlots/Plots.jl/blob/master/src/animation.jl#L66-L79 tries to avoid the imagemagick dependency, by allowing the user to execute if it is installed but not otherwise. Yet another thing that may become easier with conditional dependencies I guess. I guess there isn't another good option for doing this?

@huckl3b3rry87 this indicates your warning should be there, but maybe make it more descriptive - instead of the ubuntu-specific `apt-get` instruction say "you may also need to install the imagemagick c++ library through your operating system" after the `Pkg.add` suggestion?

**Pull request closed without merging**